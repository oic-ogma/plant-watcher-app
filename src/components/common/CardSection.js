import React from 'react';
import { StyleSheet, View } from 'react-native';

export const CardSection = props => (
  <View style={[styles.containerStyle, props.style]}>
    {props.children}
  </View>
);

const styles = StyleSheet.create({
  containerStyle: {
    borderBottomWidth: 1,
    padding: 5,
    backgroundColor: '#fff',
    justifyContent: 'flex-start',
    flexDirection: 'row',
    borderColor: '#ddd',
    position: 'relative',
  }
});
