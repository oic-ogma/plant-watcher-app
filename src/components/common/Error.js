import React from 'react';
import { Text } from 'react-native';

export const Error = ({ message }) => (
  <Text style={styles.errorTextStyle}>
    {message}
  </Text>
);

const styles = {
  errorTextStyle: {
    paddingTop: 5,
    fontSize: 18,
    alignSelf: 'center',
    color: '#FF0000',
  }
};
