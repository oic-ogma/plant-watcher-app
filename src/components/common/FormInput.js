import React from 'react';
import { FormInput } from 'react-native-elements';
import { Platform,  } from 'react-native';

export const FormInputComponent = ({ placeholder, multiline, onChangeText, value, secureTextEntry }) => {
  const { input, multilineInput } = styles;
  if (multiline) {

    return (
      <FormInput
        placeholder={placeholder}
        onChangeText={onChangeText}
        multiline={true}
        blurOnSubmit={false}
        maxHeight={150}
        inputStyle={multilineInput}
        autoGrow={true}
        value={value}
      />
    );
  }
  return (
    <FormInput
      secureTextEntry={secureTextEntry}
      onChangeText={onChangeText}
      placeholder={placeholder}
      value={value}
      inputStyle={input}
    />
  );
};

const styles = {
  multilineInput: {
    fontSize: 18,
    ...Platform.select({
      android: {
        textAlignVertical: 'top'
      }
    })
  },
  input: {
    fontSize: 18,
  }
};
