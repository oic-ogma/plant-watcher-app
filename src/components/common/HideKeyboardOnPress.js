import React from 'react';
import { TouchableOpacity, Keyboard } from 'react-native';

export const HideKeyboardOnPress = ({ children }) => (
  <TouchableOpacity onPress={Keyboard.dismiss} style={{ flex: 1 }}>
    {children}
  </TouchableOpacity>
);
