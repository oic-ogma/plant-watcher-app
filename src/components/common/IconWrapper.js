import React from 'react';
import { Icon } from 'react-native-elements';

export const IconWrapper = ({ name, type, tintColor }) => (
  <Icon
    name={name}
    type={type}
    color={tintColor}
  />
);
