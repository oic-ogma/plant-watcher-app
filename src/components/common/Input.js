import React from 'react';
import { TextInput, View, Text } from 'react-native';

const Input = ({ label, value, onChangeText, placeholder, secureTextEntry, multiline, height }) => {
  const { inputStyle, labelStyle, containerStyle } = styles;
  return (
    <View style={height ? [containerStyle, { height }] : containerStyle}>
      <Text style={labelStyle}>{label}</Text>
      <TextInput
        secureTextEntry={secureTextEntry}
        placeholder={placeholder}
        autoCorrect={false}
        style={inputStyle}
        value={value}
        onChangeText={onChangeText}
        multiline={multiline}
        underlineColorAndroid='transparent'
      />
    </View>
  );
};

const styles = {
  inputStyle: {
    color: '#000',
    paddingRight: 5,
    paddingLeft: 5,
    fontSize: 18,
    lineHeight: 23,
    flex: 5
  },
  labelStyle: {
    fontSize: 18,
    paddingLeft: 10,
    flex: 3
  },
  containerStyle: {
    height: 40,
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center'
  }
};

export { Input };
