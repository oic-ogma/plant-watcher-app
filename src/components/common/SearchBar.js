import React from 'react';
import { SearchBar } from 'react-native-elements';

const SearchBarComponent = ({ placeholder }) => {
  const { container, input, icon, clearIcon } = styles;
  return (
    <SearchBar
      lightTheme
      round
      containerStyle={container}
      inputStyle={input}
      icon={{ style: icon }}
      placeholder={placeholder}
      clearIcon={{ style: clearIcon }}
    />
  );
};

export { SearchBarComponent };

const styles = {
  container: {
    backgroundColor: '#FFF',
    borderBottomColor: '#FFF',
    borderTopColor: '#FFF'
  },
  input: {
    fontSize: 20,
    height: 50,
    paddingLeft: 45
  },
  icon: {
    top: 17.5,
    width: 30,
    fontSize: 30
  },
  clearIcon: {
    top: 17.5,
    width: 30,
    fontSize: 30
  }
};
