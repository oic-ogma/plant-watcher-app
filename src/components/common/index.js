export * from './Button';
export * from './Card';
export * from './CardSection';
export * from './Header';
export * from './Input';
export * from './Spinner';
export * from './Confirm';
export * from './Error';
export * from './ListItem';
export * from './ListArticles';
export * from './Label';
export * from './FormInput';
export * from './IconWrapper';