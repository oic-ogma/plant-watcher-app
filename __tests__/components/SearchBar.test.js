import React from 'react';
import { SearchBarComponent } from '../../src/components/common/SearchBar';
import renderer from 'react-test-renderer';

describe('サーチバー', () => {
  it('snapshot-普通', () => {
    const tree = renderer.create(<SearchBarComponent />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-valueが入力されていない', () => {
    const tree = renderer.create(<SearchBarComponent placeholder={'植物名'}/>);
    expect(tree).toMatchSnapshot();
  });

  it('snapshot-valueに値が入力されている',() => {
    const tree = renderer.create(<SearchBarComponent value={'ひまわり'} placeholder={'植物名'}/>);
    expect(tree).toMatchSnapshot();
  });
});
