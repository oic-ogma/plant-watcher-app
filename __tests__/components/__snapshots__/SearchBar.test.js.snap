// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`サーチバー snapshot-valueが入力されていない 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      Object {
        "backgroundColor": "#e1e8ee",
        "borderBottomColor": "#e1e1e1",
        "borderTopColor": "#e1e1e1",
      },
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "borderBottomColor": "rgba(0,0,0,0)",
        "borderTopColor": "rgba(0,0,0,0)",
        "flex": 0.8,
      },
    ]
  }
>
  <TextInput
    onChangeText={undefined}
    onSubmitEditing={undefined}
    placeholder="植物名"
    placeholderTextColor="#86939e"
    selectionColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        Object {
          "backgroundColor": "#bdc6cf",
        },
        false,
        Object {
          "borderRadius": 15,
        },
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "fontSize": 20,
          "height": 50,
          "paddingLeft": 45,
        },
        null,
        false,
      ]
    }
    underlineColorAndroid="transparent"
    value={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          Object {
            "fontSize": 30,
            "top": 17.5,
            "width": 30,
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`サーチバー snapshot-valueに値が入力されている 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      Object {
        "backgroundColor": "#e1e8ee",
        "borderBottomColor": "#e1e1e1",
        "borderTopColor": "#e1e1e1",
      },
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "borderBottomColor": "rgba(0,0,0,0)",
        "borderTopColor": "rgba(0,0,0,0)",
        "flex": 0.8,
      },
    ]
  }
>
  <TextInput
    onChangeText={undefined}
    onSubmitEditing={undefined}
    placeholder="植物名"
    placeholderTextColor="#86939e"
    selectionColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        Object {
          "backgroundColor": "#bdc6cf",
        },
        false,
        Object {
          "borderRadius": 15,
        },
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "fontSize": 20,
          "height": 50,
          "paddingLeft": 45,
        },
        false,
        Object {
          "paddingRight": 30,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value="ひまわり"
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          Object {
            "fontSize": 30,
            "top": 17.5,
            "width": 30,
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "right": 16,
          },
          Object {
            "fontSize": 30,
            "top": 17.5,
            "width": 30,
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;

exports[`サーチバー snapshot-普通 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#393e42",
        "borderBottomColor": "#000",
        "borderBottomWidth": 1,
        "borderTopColor": "#000",
        "borderTopWidth": 1,
      },
      Object {
        "backgroundColor": "#e1e8ee",
        "borderBottomColor": "#e1e1e1",
        "borderTopColor": "#e1e1e1",
      },
      Object {
        "backgroundColor": "rgba(0,0,0,0)",
        "borderBottomColor": "rgba(0,0,0,0)",
        "borderTopColor": "rgba(0,0,0,0)",
        "flex": 0.8,
      },
    ]
  }
>
  <TextInput
    onChangeText={undefined}
    onSubmitEditing={undefined}
    placeholder={undefined}
    placeholderTextColor="#86939e"
    selectionColor="#86939e"
    style={
      Array [
        Object {
          "backgroundColor": "#303337",
          "borderRadius": 3,
          "color": "#86939e",
          "fontSize": 17.5,
          "height": 30,
          "margin": 8,
          "overflow": "hidden",
          "paddingLeft": 26,
          "paddingRight": 19,
        },
        Object {
          "backgroundColor": "#bdc6cf",
        },
        false,
        Object {
          "borderRadius": 15,
        },
        Object {
          "backgroundColor": "rgb(255,255,255)",
          "fontSize": 20,
          "height": 50,
          "paddingLeft": 45,
        },
        null,
        false,
      ]
    }
    underlineColorAndroid="transparent"
    value={undefined}
  />
  <Text
    accessible={true}
    allowFontScaling={false}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#86939e",
          "fontSize": 16,
        },
        Array [
          Object {
            "backgroundColor": "transparent",
            "position": "absolute",
            "top": 15.5,
          },
          Object {
            "left": 16,
          },
          Object {
            "fontSize": 30,
            "top": 17.5,
            "width": 30,
          },
        ],
        Object {
          "fontFamily": "Material Icons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
